File: sanket-dalvi/Day4/longestCommonPrefix.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        def check(i):
            c = strs[0][i]
            for s in strs[1:]:
                if s[i]!=c:
                    return True
            return False
        for i in strs:
            if i == "": return ""
        if len(strs)==0:
            return ""
        elif len(strs)==1:
            return strs[0]
        p = 0
        while True:
            try:
                if check(p):
                    return strs[0][:p]
                else:
                    p += 1
            except:
                return strs[0][:p]
        
# Test
print(Solution().longestCommonPrefix(["flower","flow","flight"]))

File: parasmittal/Day 4/Longest Common Prefix.cpp
class Solution {
public:
    int findMinLen(vector<string>& strs){
        int minlen=strs[0].length();
        for(int i=1;i<strs.size();i++){
            if(minlen>strs[i].length())
                minlen=strs[i].length();
        }
        return minlen;
    }
    string longestCommonPrefix(vector<string>& strs) {
        int n=strs.size();
        if(n==0) return "";
        if(n==1) return strs[0];
        int minlen=findMinLen(strs);
        string out="";
        for(int i=0;i<minlen;i++){
            char c=strs[0][i];
            for(int j=1;j<strs.size();j++){
                if(strs[j][i]!=c)
                    return out;
            }
            out+=c;
        }
        return out;
    }
};